torch/distributed/checkpoint/_hf_storage.py

```diff
@@ -172,8 +179,14 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner) -> Future[None]:
                 for req in reqs:
                     tensor = loaded_tensors[req.dest_index.fqn]
 
-                    target_tensor = planner.resolve_tensor(req).detach()
-                    target_tensor.resize_(tensor.size())
+                    target_tensor = planner.resolve_tensor(req)
+                    if isinstance(planner, _HuggingFaceLoadPlanner) and planner.allow_tensor_resize:
```

Comment:
Is allowing tensor resizing a generic enough request that it could be part of the DefaultLoadPLanner and then you wouldn't need to check if the planner was an instance of the `_HuggingFaceLoadPlanner`?